test/unit/template_test.rb

```diff
@@ -86,23 +87,38 @@ def default_formatting
   end
 
   test "able to version a template" do
-    init_full_template(@basic_template)
-    assert_equal 0, @basic_template.version, 'expected the initial template version to be zero'
-    version2 = @basic_template.generate_version
+    template = init_full_template(@basic_template)

```

Comment:
I believe in ruby you pass the value of a reference as an argument, that means you modify @basic_template within the method (init_full_template) and return effectively a reference to the instance variable. Don't understand why do you want to assign the result into template since @basic_template contains the updates.